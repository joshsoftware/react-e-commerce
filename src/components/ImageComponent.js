import React from 'react';
import PropTypes from 'prop-types';

const ImageComponent = (props) => {
  const { style, src, className, alt } = props;

  return <img style={style} src={src} className={className} alt={alt} />;
};

export default ImageComponent;

ImageComponent.propTypes = {
  style: PropTypes.object.isRequired,
  src: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired
};
