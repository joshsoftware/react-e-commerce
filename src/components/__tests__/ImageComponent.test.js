import ImageComponent from '../ImageComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('Image Component', () => {
  it('Must return Image', () => {
    const style = { width: '50px' };
    const src = "https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png";
    const className = "rounded";
    const alt = "img";
    const { asFragment } = render(<ImageComponent style={style} src={src} className={className} alt={alt}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});