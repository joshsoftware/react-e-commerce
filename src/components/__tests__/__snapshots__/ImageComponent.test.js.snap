// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component Must return Image 1`] = `
<DocumentFragment>
  <img
    alt="img"
    class="rounded"
    src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
    style="width: 50px;"
  />
</DocumentFragment>
`;
